### DISCUSSION TITLE
Manage tui-loader with interceptor

### DISCUSSION THREAD

**Discussion by luca-sacco** [taiga-family/taiga-ui#1293] [Q&A] [answered]
Hi everyone,
I wanted to centralize the hide and show logic of the tui-loader going to intercept (via interceptor) all the http requests, this to avoid managing the loading at each http request, on all the different parts / components. What do you think about it? The problem I have is in the case of multiple http requests where the concurrency should be handled (the loading appears and disappears continuously for each http request).
Any suggestions?

**Comment by waterplea** [comment, marked as answer]
You can probably create a global singleton service that takes care of it.

**Reply by luca-sacco** [reply]
So you suggest that the interceptor calls some methods, for example, show() hide(), of a global sigleton where a queue of http requests is managed inside this singleton?

**Reply by waterplea** [reply]
You can make a service with `loading$: Observable<boolean>` that would count the number of ongoing requests and when this number is 0 emit `false`, otherwise emit `true`.
